edk2toolext/environment/repo_resolver.py

```diff
@@ -261,7 +262,20 @@ def repo_details(abs_file_system_path: os.PathLike) -> dict:
             details["Head"] = {"HexSha": str(repo.head.commit.hexsha), "HexShaShort": str(repo.head.commit.hexsha[:7])}
             details["Valid"] = True
             details["Bare"] = repo.bare
-            details["Dirty"] = repo.is_dirty(untracked_files=True)
+            retry_attempts = 5
+            for attempt in range(retry_attempts):
```

Comment:
can you pull this out into a separate function? something like `git_retry(lamda, attempts) -> lambda_output` This could be useful down the road for other git commands, and also keeps this function clean.

Reply:
Done